src/parser/warrior/fury/modules/talents/SuddenDeath.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import { formatPercentage, formatThousands } from 'common/format';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import Analyzer from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import { SELECTED_PLAYER } from 'parser/core/EventFilter';
+
+/**
+ * Your attacks have a chance to reset the cooldown of Execute
+ * and make it usable on any target, regardless of their health.
+ */
+
+class SuddenDeath extends Analyzer {
+  suddenDeathProcs = 0;
+  suddenDeathProcsUsed = 0;
+  suddenDeathDamage = 0;
+
+  lastExecuteCast = 0;
+  lastSuddenDeathExecuteCast = 0;
+
+  constructor(...args) {
+    super(...args);
+
+    this.active = this.selectedCombatant.hasTalent(SPELLS.SUDDEN_DEATH_TALENT_FURY.id);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EXECUTE_FURY), this.onExecuteCast);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell([SPELLS.EXECUTE_DAMAGE_FURY, SPELLS.EXECUTE_DAMAGE_OH_FURY]), this.onExecuteDamage);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.SUDDEN_DEATH_TALENT_FURY_BUFF), this.onSuddenDeathProc);
+    this.addEventListener(Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.SUDDEN_DEATH_TALENT_FURY_BUFF), this.onSuddenDeathProc);
+  }
+
+  onExecuteCast(event) {
+    if (this.selectedCombatant.hasBuff(SPELLS.SUDDEN_DEATH_TALENT_FURY_BUFF.id)) {
+      this.lastSuddenDeathExecuteCast = event.timestamp;
+      this.suddenDeathProcsUsed += 1;
+    }
+    this.lastExecuteCast = event.timestamp;
+  }
+
+  onExecuteDamage(event) {
+    if (this.lastExecuteCast === this.lastSuddenDeathExecuteCast) {
```

Comment:
You actually only need to check on the damage event to see if the player had the buff a few milliseconds ago, rather than comparing it to the cast event. Technically dont need to listen to the cast event at all :)

Reply:
So I found out about the buffer duration on the hasBuff function, and that could be a good way to go with this, but I need to use the cast listener to check how many of the procs were used as the damage can be cleaved onto 4 targets, and we only want to count the one proc being used. 

Also, to check the number of uses against high hp targets, we need to do that on cast, as you could just use execute on a low hp target next to a high hp one and have that damage cleave.